lib/events/auditwriter.go

```diff
@@ -371,8 +371,9 @@ func (a *AuditWriter) Close(ctx context.Context) error {
 // Complete closes the stream and marks it finalized,
 // releases associated resources, in case of failure,
 // closes this stream on the client side
-func (a *AuditWriter) Complete(ctx context.Context) error {
-	return a.Close(ctx)
+func (a *AuditWriter) Complete(ctx context.Context) (*UploadMetadata, error) {
+	m, _ := a.stream.Complete(ctx)
```

Comment:
I think that could be a good idea. So should we should emit the event from `processEvents()`

Reply:
No, my idea was just store it in an attribute on the audit writer but there could be a better way - will think about it overnight.

Reply:
If we store in `AuditWriter` instead of passing the upload data back through the `Complete()` method, I am pretty sure the the event would need to be emitted from elsewhere (maybe `processEvents`). Is there a reason why it would be a good idea to store `UploadMetaData` in `AuditWriter`?

Reply:
Yes, I think this change will break Audit writer logic. Why not emit event on the auth server grpc side?